service/src/main/kotlin/io/provenance/api/frameworks/objectStore/ObjectStoreService.kt

```diff
@@ -1,57 +1,80 @@
 package io.provenance.api.frameworks.objectStore
 
-import com.google.protobuf.Message
 import io.provenance.api.domain.objectStore.ObjectStore
-import io.provenance.api.frameworks.config.ObjectStoreConfig
+import io.provenance.api.frameworks.config.ObjectStoreProperties
 import io.provenance.api.models.eos.store.StoreProtoResponse
 import io.provenance.api.models.eos.store.toModel
+import io.provenance.api.util.toModel
 import io.provenance.scope.encryption.crypto.Pen
 import io.provenance.scope.encryption.domain.inputstream.DIMEInputStream
 import io.provenance.scope.encryption.ecies.ProvenanceKeyGenerator
 import io.provenance.scope.encryption.model.DirectKeyRef
 import io.provenance.scope.objectstore.client.OsClient
-import java.io.InputStream
+import java.io.ByteArrayInputStream
+import java.security.KeyPair
 import java.security.PrivateKey
 import java.security.PublicKey
+import java.time.Duration
 import java.util.concurrent.TimeUnit
+import org.apache.commons.codec.binary.Base64
 import org.springframework.stereotype.Component
+import tech.figure.objectstore.gateway.client.GatewayClient
+import tech.figure.objectstore.gateway.client.GatewayJwt
 
 @Component
 class ObjectStoreService(
-    private val osConfig: ObjectStoreConfig
+    private val objectStoreProperties: ObjectStoreProperties,
 ) : ObjectStore {
     // Retrieve the asset as a byte array and decrypt using the provided keypair
-    override fun retrieveAndDecrypt(
-        client: OsClient,
-        hash: ByteArray,
+    override fun <T> retrieveAndDecrypt(
+        client: T,
+        hash: String,
         publicKey: PublicKey,
         privateKey: PrivateKey,
-    ): ByteArray {
-        val future = client.get(hash, publicKey)
-        val res: DIMEInputStream = future.get(osConfig.timeoutMs, TimeUnit.MILLISECONDS)
-        return res.getDecryptedPayload(DirectKeyRef(publicKey, privateKey)).readAllBytes()
-    }
+    ): ByteArray =
+        when (client) {
+            is OsClient -> {
```

Comment:
Not a huge fan of doing it this way with `when` and `else -> error`, but I understand why you did it. Fine with it as long as there are only two possible client classes in play.

Reply:
ya, in this case only two clients